Rust parsing error on saves with accented names; also with C.A.R.E. country

Commit a8bab1ec402fabb7c905fcda9ab5a00234f833d3 breaks parsing a couple saves of mine, in the Rust portion.

1. Can't handle `á` in names
2. Can't handle a name of `=` (see first comment)

3.10.4 save (Linux; from Observer mode)(rename to .sav):

[kaanvisamfriendship3_2508.07.01.zip](https://github.com/eliasdoehne/stellaris-dashboard/files/13707663/kaanvisamfriendship3_2508.07.01.zip)

Old save (original report; I don't believe it was in Observer mode):

[nexitronawareness4_2948.03.14.zip](https://github.com/eliasdoehne/stellaris-dashboard/files/13663685/nexitronawareness4_2948.03.14.zip):

Repro - using docker for isolation

`docker run -it forbiddenlake/stellaris-dashboard:v6-debian-bookworm-py3.11 bash`

In this container, run:

```sh
pip install -e .
curl -sSf https://sh.rustup.rs | sh -s -- -y && . "$HOME/.cargo/env"
rm test/saves/nexitronawareness_1329922464/2565.06.02.sav
git checkout a8bab1ec402fabb7c905fcda9ab5a00234f833d3
```

Outside the container (in another window), copy the offending save in (remember to rename it) with something like the following, where 98a4b7dd6cb4 is the id of the container you just ran:

`docker cp /syncthing/stellaris-saves/nexitronawareness4_-920922254/2948.03.14.sav  98a4b7dd6cb4:/home/stellaris/stellaris-dashboard/test/saves/nexitronawareness_1329922464/`

Back in the container, run:

```sh
pytest test/parser_test.py -k "test_real_save" -s
```

This fails with: 

`FAILED test/parser_test.py::test_real_save - ValueError: Failed to parse /home/stellaris/stellaris-dashboard/test/saves/nexitronawareness_1329922464/2948.03.14.sav: Failed to parse save gamestate`

Next, check out the previous commit and rebuild the rust parser:

```sh
git checkout 00b4f03b19c6bb51cf6cfae907a8edbf1c875c81
cd stellarisdashboard/parsing/rust_parser
env VIRTUAL_ENV=$(python3 -c 'import sys; print(sys.base_prefix)') maturin develop -r
cd -
```

And rerun:

```sh
pytest test/parser_test.py -k "test_real_save" -s
```

The parsing succeeds.

Aside: A later commit (7cabf081af69bb8367e4ec1cabba551e39851bcc) contains updates that will prevent parsing this save file for a different reason, which is fixed in #124.

I don't know Rust at all to debug this further, so I'm submitting this. cc @MichaelMakesGames 

People do run old versions of the game, so I think keeping backwards compatibility is good if feasible.

--- a/stellarisdashboard/parsing/rust_parser/src/parser.rs
+++ b/stellarisdashboard/parsing/rust_parser/src/parser.rs
@@ -250,6 +250,7 @@ fn parse_str(input: &str) -> IResult<&str, &str> {
 
 fn _is_valid_unquoted_str_char(c: char) -> bool {
     !c.is_whitespace() &&
+    c != '"' &&
     c != '=' &&
     c != '{' &&
     c != '}' &&
@@ -279,7 +280,7 @@ fn parse_color(input: &str) -> IResult<&str, (&str, f64, f64, f64)> {
 
 #[allow(dead_code)]
 pub fn debug_str(input: &str) -> () {
-    let prefix = &input[..min(input.len(), 150)];
+    let prefix: String = input.chars().take(150).collect();
     println!("{:?}", prefix);
     println!("{}", prefix);
 }

